src/pkg/cli/getServices.go

```diff
@@ -62,5 +70,44 @@ func GetServices(ctx context.Context, projectName string, provider client.Provid
 		servicesResponse.Services[i] = nil
 	}
 
-	return term.Table(printServices, "Service", "Deployment", "PublicFqdn", "PrivateFqdn", "Status")
+	return term.Table(printServices, "Service", "Deployment", "PublicFqdn", "PrivateFqdn", "State")
+}
+
+func UpdateServiceStates(ctx context.Context, serviceInfos []*defangv1.ServiceInfo) {
+	// Create a context with a timeout for HTTP requests
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+	var wg sync.WaitGroup
+
+	for _, serviceInfo := range serviceInfos {
+		for _, endpoint := range serviceInfo.Endpoints {
+			if !strings.Contains(endpoint, ":") {
+				wg.Add(1)
+				go func(serviceInfo *defangv1.ServiceInfo) {
+					defer wg.Done()
+					url := "https://" + endpoint + "/" // TODO: use serviceInfo.Healthcheck
+					// Use the regular net/http package to make the request without retries
+					req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+					if err != nil {
+						term.Errorf("Failed to create healthcheck request for %q at %s: %s", serviceInfo.Service.Name, url, err.Error())
+						return
+					}
+					term.Debugf("[%s] checking health at %s", serviceInfo.Service.Name, url)
+					resp, err := http.DefaultClient.Do(req)
+					if err != nil {
+						term.Errorf("Healthcheck failed for %q at %s: %s", serviceInfo.Service.Name, url, err.Error())
+						return
+					}
+					defer resp.Body.Close()
+					if resp.StatusCode >= 200 && resp.StatusCode < 300 {
+						serviceInfo.State = defangv1.ServiceState_DEPLOYMENT_COMPLETED
+						term.Debugf("[%s] ✔ healthy", serviceInfo.Service.Name)
+					} else {
+						term.Debugf("[%s] ✘ unhealthy (status %d)", serviceInfo.Service.Name, resp.StatusCode)
```

Comment:
Status has more info than StatusCode
```suggestion
						term.Debugf("[%s] ✘ unhealthy (status %s)", serviceInfo.Service.Name, resp.Status)
```